File: android/app/src/main/java/io/storj/mobile/storjlibmodule/dataprovider/contracts/SynchronizationQueueContract.java
package io.storj.mobile.storjlibmodule.dataprovider.contracts;

public final class SynchronizationQueueContract extends BaseContract {
    public final static String TABLE_NAME = "synchronizationQueue";

    public final static String _FILE_NAME = "fileName";
    public final static String _LOCAL_PATH = "localPath";
    public final static String _STATUS = "status";
    public final static String _ERROR_CODE = "errorCode";
    public final static String _SIZE = "size";
    public final static String _COUNT = "count";
    public final static String _CREATION_DATE = "creationDate";

    public final static String _BUCKET_ID = "bucketId";
    public final static String _FILE_HANDLE = "fileHandle";

    public static String createTable() {
        return String.format("create table if not exists %s (" + //table
                        "%s INTEGER primary key AUTOINCREMENT not null, " + //id NOT SET
                        "%s TEXT not null, " + //fileName
                        "%s TEXT not null, " + //localPath
                        "%s INTEGER default 0, " + //status
                        "%s INTEGER default 0, " + //errorCode
                        "%s INTEGER default 0, " + //size
                        "%s INTEGER default 0, " + //count
                        "%s TIMESTAMP DEFAULT CURRENT_TIMESTAMP, " + //creationDate //NOT SET
                        "%s TEXT not null, " + //bucketId
                        "%s INTEGER, " + //fileHandle
                        "FOREIGN KEY(%s) REFERENCES uploadingFiles(%s) ON DELETE SET NULL, " + //uploadingfiles FK
                        "FOREIGN KEY(%s) REFERENCES buckets(%s) ON DELETE SET NULL)", //buckets FK
                TABLE_NAME, _ID, _FILE_NAME, _LOCAL_PATH, _STATUS, _ERROR_CODE, _SIZE, _COUNT, _CREATION_DATE, _BUCKET_ID, _FILE_HANDLE, _FILE_HANDLE, _ID, _BUCKET_ID, _ID);
    }
}

File: ios/dataProvider/contracts/SynchronizationQueueContract.m
//
//  SynchronizationQueueContract.m
//  StorjMobile
//
//  Created by Developer Mac on 31.05.2018.
//  Copyright © 2018 Storj. All rights reserved.
//

#import <Foundation/Foundation.h>
#import "SynchronizationQueueContract.h"

@implementation SynchronizationQueueContract

static NSString * const _TABLE_NAME = @"SynchronizationQueue";

static NSString * const _ID = @"_id";
static NSString * const _FILE_NAME = @"fileName";
static NSString * const _LOCAL_PATH = @"localPath";
static NSString * const _STATUS = @"status";
static NSString * const _ERROR_CODE = @"errorCode";
static NSString * const _SIZE = @"size";
static NSString * const _COUNT = @"count";
static NSString * const _CREATION_DATE = @"creationDate";

static NSString * const _FILE_HANDLE = @"fileHandle";
static NSString * const _BUCKET_ID = @"bucketId";

+(NSString *) TABLE_NAME
{
  return _TABLE_NAME;
}

+(NSString *) ID
{
  return _ID;
}

+(NSString *) FILE_NAME
{
  return _FILE_NAME;
}

+(NSString *) LOCAL_PATH
{
  return _LOCAL_PATH;
}

+(NSString *) STATUS
{
  return _STATUS;
}

+(NSString *) ERROR_CODE
{
  return _ERROR_CODE;
}

+(NSString *) SIZE
{
  return _SIZE;
}

+(NSString *) COUNT
{
  return _COUNT;
}

+(NSString *) CREATION_DATE
{
  return _CREATION_DATE;
}

+(NSString *) FILE_HANDLE
{
  return _FILE_HANDLE;
}

+(NSString *) BUCKET_ID
{
  return _BUCKET_ID;
}

+(NSString *) createTable
{
  return [NSString stringWithFormat: @"create table if not exists %@ (\
%@ INTEGER primary key AUTOINCREMENT not null, \
%@ TEXT not null, \
%@ TEXT not null, \
%@ INTEGER default 0, \
%@ INTEGER default 0, \
%@ INTEGER default 0, \
%@ INTEGER default 0, \
%@ TIMESTAMP default CURRENT_TIMESTAMP, \
%@ TEXT not null, \
%@ INTEGER, \
FOREIGN KEY(%@) REFERENCES uploadingFiles(%@) ON DELETE CASCADE, \
FOREIGN KEY(%@) REFERENCES buckets(%@) ON DELETE CASCADE);",
          _TABLE_NAME, _ID, _FILE_NAME, _LOCAL_PATH, _STATUS, _ERROR_CODE, _SIZE, _COUNT,
          _CREATION_DATE, _BUCKET_ID, _FILE_HANDLE, _FILE_HANDLE, _FILE_HANDLE, _BUCKET_ID, _ID];
}

@end


